From d300476c9515a3e1c6a9b66a3a4f430b7395f4e5 Mon Sep 17 00:00:00 2001
From: Rob Vogelaar <revogelaar@gmail.com>
Date: Fri, 26 Dec 2025 15:19:25 -0800
Subject: [PATCH] patch

Signed-off-by: Rob Vogelaar <revogelaar@gmail.com>
---
 src/rtmessage/rtrouted.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/rtmessage/rtrouted.c b/src/rtmessage/rtrouted.c
index 9879e9d..f406b20 100644
--- a/src/rtmessage/rtrouted.c
+++ b/src/rtmessage/rtrouted.c
@@ -416,9 +416,11 @@ rtRouted_AddAlias(char const* exp, rtRouteEntry * route)
   rtError rc = RT_OK;
   rtLog_Debug("AddAlias route=[%p] address=[%s] expression=[%s] alias=[%s]", route, route->subscription->client->ident, route->expression, exp);
   rc = rtRoutingTree_AddTopicRoute(gRoutingTree, exp, (void *)route, 1/*error if duplicate entry*/);
-  if (RT_ERROR_DUPLICATE_ENTRY == rc)
+  if (RT_ERROR_DUPLICATE_ENTRY == rc) {
     if (rtRouted_ShouldLimitLog(exp))
       rtLog_Warn("Rejecting Duplicate Registration of [%s] by [%s] thro [%s]", exp, route->expression, route->subscription->client->ident);
+    rc = RT_OK;
+  }
 
   return rc;
 }
-- 
2.25.1

